docs/articles/rag-application-communication-system.md

```diff
@@ -0,0 +1,194 @@
+
+### 3.2 Managing data flows with classifiers
+Fortunately, we have the best tools to date for processing bad (and good) data: language models, despite their limitations for search, are amazing at data processing. I mean here language models in general, not just the popular decoders: the previous generation of encoders and encoder-decoders currently enjoy a much needed revival; they provide a much better parameters / performance / robustness ratio than GPT-4, Llama or Mistral for many data tasks.
+
+I increasingly rely on classifiers or reformulators (for want of better name) to build RAG systems. This includes:
+
+- * Automated re-routing of user queries. A major issue of more generalist chat systems is deciding 1) whether or not to activate the retrieval of external resources 2) which collection of resources to prioritize (maybe not all are needed; reducing volume can relieve stress to the system) 3) when a new round of retrieval becomes necessary (e.g., when the topic of a conversation has changed).
+- * Query expansion. LLM and embedding models, because they're trained primarily on complete sentences, struggle with unstructured queries. You may require further contextualization by adding elements (e.g., past queries).
+- * Text segmentation (rather than just “chunking”). Proper segmentation has proven to be one of the most powerful performance boosts. But as user traffic increases, and more document coverage is needed, iteratively expanding your RAG system (typically by parsing different sets of resources with existing ones) can become prohibitively costly. Lately, I’ve been experimenting a lot with agnostic encoder models for token-classification tasks, able to reconstruct text parts (including titles, bibliography) from raw text comprehension, regardless of the original format.
+
+For automated query rerouting, query expansion, and text segmentation, using an LLM is overkill and results in performance lag and degradation. Once properly re-trained, encoder models yield comparable results, are very fast, not expensive to host, and better fitted to the overall purpose of classification. In addition, encoders don't require structured generation, regex, or data constraints - the data design is already built-in.
+
+### 3.3 An alternative to generation: synthetic data curation
+I’m expecting that a major direction of research in the months to come will be pre-generation of system output and input - i.e., basically, swapping models in production with optimized retrieval of synthetic data. This was the approach retained by Klarna for their integration of ChatGPT. It yields obvious advantages in terms of robustness and security (prompt injection is simply impossible).
+
```

Comment:
@morkapronczay
As I understand it, Klarna collab'ed with OpenAI to build an integrated plugin to chatGPT... rather than swapping models?

Reply:
So for Klarna, haven't been able to relocate the source, so you can drop the mention.
Suggested rewrite: "basically, relying on the retrieval of pre-generated outputs from a model, rather than generations at inference time."